components/server/src/api/workspace-service-api.ts

```diff
@@ -92,4 +99,80 @@ export class WorkspaceServiceAPI implements ServiceImpl<typeof WorkspaceServiceI
         response.pagination.total = resultTotal;
         return response;
     }
+
+    async createAndStartWorkspace(req: CreateAndStartWorkspaceRequest): Promise<CreateAndStartWorkspaceResponse> {
+        // We rely on FGA to do the permission checking
+        if (req.source?.case !== "contextUrl") {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "not implemented");
```

Comment:
Just noticed we don't have `501 - not implemented` yet... not sure if it's worth introducing here. But feels more correct. :shrug: :balance_scale:

Reply:
yes, it would be good to map to gRPC unimplemented